### PR TITLE
fix handling of the deprecated autoscaling settings

### LINKED ISSUE #41011: Istio installation using IOP failed when hpaSpec for cpu utilization configured  for components
Installing istio with IOP(default profile) failed when hpaSpec for cpu utilization configured.

Error message - 
```
- Processing resources for Istio core.
   ✔ Istio core installed
   - Processing resources for Istiod.2022-08-30T23:43:32.119129Z        error   installer       failed to update resource with server-side apply for obj HorizontalPodAutoscaler/istio-system/istiod: failed to create typed patch object (istio-system/istiod; autoscaling/v2beta1, Kind=HorizontalPodAutoscaler): .spec.metrics[0].resource.target: field not declared in schema
   
   ✘ Istiod encountered an error: failed to update resource with server-side apply for obj HorizontalPodAutoscaler/istio-system/istiod: failed to create typed patch object (istio-system/istiod; autoscaling/v2beta1, Kind=HorizontalPodAutoscaler): .spec.metrics[0].resource.target: field not declared in schema
```

**Istio version** - 1.14.3
**Kubernetes version** - 1.24.3

**Attachments**: Istio IOP file generated using `istioctl profile dump`
[istio_1.14.3_default_profile.txt](https://github.com/istio/istio/files/9577498/istio_1.14.3_default_profile.txt)

### PR DESCRIPTION
**Pull request by richardwxn**
Fix: #41011 

The root cause is that: 
1. we recently made a change which does #37872 : select the API version of the templates based on the values.autoscalingv2 field, which is set to false if any of the deprecated autoscaling fields are detected in the K8S sections of IOP during the K8S overlay logic, 
2. however the legacy logic in the translate_values.go would also override the K8S sections of the IOP by doing a reverse translation from .values section, which resulted in an inconsistency, i.e. using v2 fields under v2beta1 template. Actually, this part of logic is not needed anymore, we can just remove that and pass it as it is in .values section.

Actually a bunch of similar conversion in the translate_values.go are not needed anymore, I would remove them later as well.

Another possible fix is just to convert the deprecated settings to the corresponding fields in v2 during installation code, all of the deprecated settings actually have corresponding fields in new API(but different locations), so that we can just get rid of the v2beta1 template part entirely. But it may cause confusion if users have direct advanced K8S override that manipulate based on the old API paths

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
Would maybe be nice to have the issue as a test case to prevent regressions?

**Comment by zirain** [comment]
> Would maybe be nice to have the issue as a test case to prevent regressions?

test case from https://github.com/istio/istio/pull/41029 can be used. not 100% sure.

File: operator/pkg/translate/translate_value.go [hunk 1]
```diff
@@ -560,10 +559,6 @@ func (t *ReverseTranslator) translateK8sTree(valueTree map[string]any,
 			k8sSettingName = path[len(path)-1]
 		}
 		if k8sSettingName == "autoscaleEnabled" {
-			if err := translateHPASpec(inPath, v.OutPath, valueTree, cpSpecTree); err != nil {
```
Comment (zirain) [inline operator/pkg/translate/translate_value.go:563]: this looks better than https://github.com/istio/istio/pull/41029

**Comment by richardwxn** [comment]
@howardjohn @ostromart @zirain added two test cases

**Review by howardjohn — APPROVED** [review]
:+1: :+1:

**Pull request merged**